packages/visual-editor/src/components/Dropzone/Hitboxes.tsx

```diff
@@ -0,0 +1,206 @@
+            top: bottom + calcOffsetDepth(zoneDepth),
+            left,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.LEFT:
+          return {
+            width: HITBOX_WIDTH,
+            height: height - HITBOX_HEIGHT,
+            left: left - calcOffsetDepth(zoneDepth) - HITBOX_WIDTH / 2,
+            top: top + HITBOX_HEIGHT / 2,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.RIGHT:
+          return {
+            width: HITBOX_WIDTH,
+            height: height - HITBOX_HEIGHT,
+            left: right - calcOffsetDepth(zoneDepth) - HITBOX_WIDTH / 2,
+            top: top + HITBOX_HEIGHT / 2,
+            zIndex: 100 + zoneDepth,
+          };
+        case HitboxDirection.SELF_VERTICAL: {
+          if (height > 60 && !isDeepZone) {
+            return { display: 'none' };
+          }
+
+          const selfHeight = isDeepZone ? MIN_DEPTH : MIN_HEIGHT;
+          return {
+            width,
+            height: selfHeight,
+            left,
+            top: top + height / 2 - selfHeight / 2,
+            zIndex: 1000 + zoneDepth,
+          };
+        }
+        case HitboxDirection.SELF_HORIZONTAL: {
+          if (width > DRAGGABLE_WIDTH) {
+            return { display: 'none' };
+          }
+          return {
+            width: width - DRAGGABLE_WIDTH * 2,
+            height,
+            left: left + (DRAGGABLE_WIDTH * 2) / 2,
+            top,
+            zIndex: 1000 + zoneDepth,
+          };
+        }
+        default:
+          return {};
+      }
+    },
+    [zoneDepth, domRect],
+  );
+
+  const ActiveHitboxes = (
+    <>
+      {isVertical ? (
+        <div
+          data-ctfl-zone-id={zoneId}
+          className={styles.hitbox}
+          style={getStyles(HitboxDirection.SELF_VERTICAL)}
+        />
+      ) : (
+        <div
+          data-ctfl-zone-id={zoneId}
+          className={styles.hitbox}
+          style={getStyles(HitboxDirection.SELF_HORIZONTAL)}
+        />
+      )}
+      {isRoot && enableRootHitboxes ? (
+        <div
+          data-ctfl-zone-id={parentZoneId}
+          className={styles.hitbox}
+          style={getStyles(HitboxDirection.BOTTOM)}
+        />
+      ) : isVertical ? (
+        <>
+          <div
+            data-ctfl-zone-id={parentZoneId}
+            className={styles.hitbox}
+            style={getStyles(HitboxDirection.TOP)}
+          />
+          <div
+            data-ctfl-zone-id={parentZoneId}
+            className={styles.hitbox}
+            style={getStyles(HitboxDirection.BOTTOM)}
+          />
+        </>
+      ) : (
+        <>
+          <div
+            data-ctfl-zone-id={parentZoneId}
+            className={styles.hitbox}
+            style={getStyles(HitboxDirection.LEFT)}
+          />
+          <div
+            data-ctfl-zone-id={parentZoneId}
+            className={styles.hitbox}
+            style={getStyles(HitboxDirection.RIGHT)}
+          />
+        </>
+      )}
```

Comment:
Some repetition could be eliminated here as well since the only difference is the style prop value

```ts
      ) : (
        <>
          <div
            data-ctfl-zone-id={parentZoneId}
            className={styles.hitbox}
            style={getStyles(isVertical ? HitboxDirection.TOP : HitboxDirection.LEFT)}
          />
          <div
            data-ctfl-zone-id={parentZoneId}
            className={styles.hitbox}
            style={getStyles(isVertical ? HitboxDirection.BOTTOM : HitboxDirection.RIGHT)}
          />
        </>
      )}
```

Reply:
You're officially the king of ternary optimizations 😎 👑